### ISSUE TITLE
[BUG] MaskedDataset._mask_rna masks padding positions when last non-padding token is masked

### ISSUE THREAD

**Issue by officialasishkumar** [gc-os-ai/pyaptamer#562]
LLM generated content, by Claude (Opus 4.7)

**Describe the bug**

When `MaskedDataset` is constructed with `is_rna=True`, the `_mask_rna` helper masks "adjacent" positions of each masked token to account for base pairing. The bounds check uses `self.max_len`, but `max_len` is the *padded* length, not the actual non-padding length of a given sequence. As a result, when the last non-padding token at index `seq_len - 1` is selected for masking, the position `seq_len` (the first padding slot) is also overwritten with `mask_idx`. This corrupts the input by turning a padding token into a "real" masked position the model is then asked to predict against.

The `__getitem__` already filters mask candidates to non-padding positions via `valid_positions = self.box[x_masked > 0]`, so this side effect is purely a bug in `_mask_rna`'s bounds.

**To Reproduce**

```python
import random
from pyaptamer.datasets.dataclasses import MaskedDataset

# 3 real tokens followed by 2 padding zeros
sequences = [[1, 2, 3, 0, 0]]
targets = [[1, 2, 3, 0, 0]]

ds = MaskedDataset(
    sequences, targets, max_len=5, mask_idx=99, masked_rate=1.0, is_rna=True
)

random.seed(0)
x_masked, y_masked, x, y = ds[0]
print(x_masked.tolist())
# observed: [99, 99, 99, 99, 0]
# index 3 was a padding token in the original input but is now mask_idx=99
```

**Expected behavior**

Padding positions must never be overwritten with `mask_idx`. Adjacent positions for RNA base-pair masking should only be marked when they fall within the actual sequence (`pos < seq_len - 1` for the right neighbour, with the existing `pos > 0` for the left neighbour).

After the fix, the same reproducer yields `[99, 99, 99, 0, 0]` (or another permutation that leaves the padding zeros untouched).

**Additional context**

The `__getitem__` method already computes `seq_len = torch.sum(x_masked > 0)` for the random sampling step, so the value is readily available and just needs to be propagated to `_mask_rna`.

This is independent of the open work on `MaskedDataset` random replacement (issue #500 / PR #502), which concerns the BERT-style 10 percent random-token branch rather than the RNA adjacent masking path.

**Versions**

```
pyaptamer 0.1.0a1 (main, commit fd41ee6)
```
